vmm/src/vm.rs

```diff
@@ -587,7 +585,7 @@ impl Vm {
             seccomp_action.clone(),
             vm_ops,
             #[cfg(feature = "tdx")]
-            tdx_enabled,
+            config.lock().unwrap().tdx.is_some(),
```

Comment:
You end up locking config twice here as a parameters (for the cpus and the tdx state) which will lead to a deadlock at runtime. This is why it is a separate variable.

Reply:
Got you,let's close this pr